vuu-ui/packages/vuu-utils/src/logging-utils.ts

```diff
@@ -9,68 +11,111 @@ export interface TableLogFn {
 	(properties?: object): void
 }
 
+type loggingSettings = {
+	loggingLevel: string;
+}
+
 export interface Logger {
-	log: LogFn;
 	warn: LogFn;
 	error: LogFn;
-	group: LogFn;
-	groupCollapsed: LogFn;
-	groupEnd: LogFn;
-	assert: AssertLogFn;
-	trace: LogFn;
 	debug: LogFn;
 	info: LogFn;
-	table: TableLogFn
 }
 
+export type LogLevel = 'debug' | 'info' | 'warn' | 'error';
+
 export type BuildEnv = 'production' | 'development';
-// eslint-disable-next-line @typescript-eslint/no-empty-function
-const NO_OP: LogFn = () => {};
+
+const NO_OP = () => undefined;
+
+export const loggerNew = (category:string, logLevel: string) => {
+	const errorEnabled = logLevel === 'error';
+	const warnEnabled = errorEnabled || logLevel === 'warn';
+	const infoEnabled = errorEnabled || warnEnabled || logLevel === 'info';
+	const debugEnabled = errorEnabled || warnEnabled || infoEnabled || logLevel === 'debug';
```

Comment:
the logic here is wrong, see the pseudocode in the linked issue.
For example, warning should not be enabled just because error is.
The logging levels form a hierarchy. If a particular level is set , all levels above are set.
"debug" is always the lowest level, with "debug" set you get ALL the log levels. If "info" is set, you get all logging except debug messages etc.